### PR TITLE
get_all and get_[region] fixed.

### PR DESCRIPTION
**Pull request by samosaara**
Some more bug fixes from my #70 disaster that were discovered in #82. That's why we needed unit testing in the first place, feeling real guilty, and sorry :sweat:  This PR also includes the changes in #85.

### CODE REVIEW

File: database.py [hunk 1]
```diff
@@ -230,7 +230,7 @@ def __load_data(self):
 
     def __load_extra(self):
         """Load all the file names of the images in the Extra folder."""
-        for file in os.listdir(self.directory + "/./Images/Extra"):
+        for file in os.listdir(self.directory + "/Images/Extra"):
             if file.endswith(".jpg"):
                 name = os.path.join("/Images/Extra", file)\
                         .split('/')[-1][0:-4].lower()
```
Comment (cclauss) [inline database.py]: This looks great!  It is good to see that that tests are starting to help.

Could you please replace lines 233 to 236 with something like:
```python
extra_dir = os.path.join(self.directory, "Images", "Extra")
for file in os.listdir(extra_dir):
    name, ext in os.path.splitext(file.lower())
    if ext == ".jpg":
```

The code that is currently on lines 235 and 236 makes my head spin.

File: test_main.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def test_no_args(capsys):
 def test_len():
     # Database unfortunately makes db.__MAX_ID private :-(
     __MAX_ID = 493
-    assert len(db) == __MAX_ID + len(db.get_extra())
+    assert len(db.get_all()) == __MAX_ID + len(db.get_extra())
```
Comment (cclauss) [inline test_main.py]: Let's keep **both lines** here.  The red line tests the Database.__len__() and the red line tests db.get_all().  Both are valid and important tests.  The more tests helps find corner cases.

File: database.py [hunk 3]
```diff
@@ -230,11 +230,12 @@ def __load_data(self):
...
-            if file.endswith(".jpg"):
-                name = os.path.join("/Images/Extra", file)\
-                        .split('/')[-1][0:-4].lower()
-                path = self.directory + "/Images/Extra/" + name + ".jpg"
+        extra_dir = os.path.join(self.directory, "Images", "Extra")
+        for file in os.listdir(extra_dir):
+            name, ext = os.path.splitext(file.lower())
+            if ext == '.jpg':
+                path = os.path.join(self.directory, 'Images', 'Extra',
+                                    name + ext)
```
Comment (cclauss) [inline database.py]: path = os.path.join(extra_dir, file)

Comment (samosaara) [inline database.py]: :man_facepalming:

**Comment by cclauss** [comment]
So... This brings us back to a basic question...  When I do `Database().get_johto()` should I expect both the non-extra **and** the extra pokemon or should I **only** expect the non-extra pokemon?

https://travis-ci.org/cclauss/flake8_for_pt still shows errors but they are probably more about my testing assumptions.

File: database.py [hunk 4]
```diff
@@ -144,7 +144,7 @@ def get_dark(self, threshold=0.6, all_pkmn=False):
     def __get_region(self, region):
         # Helper method for getting all the Pokemon of a specified region.
         return [pokemon for pokemon in self.__pokemon_list
-                if pokemon.get_region() == region]
+                if pokemon.get_region() == region and not pokemon.is_extra()]
```
Comment (samosaara) [inline database.py]: As you can see here I explicitly excluded extra pokemon from the `get_region` methods.

**Comment by cclauss** [comment]
Understood I will adjust tests.  What do you call the non-extra pokemon?!?  Standard, built-in, normal, original?

**Comment by samosaara** [comment]
Normal can be ambiguous because is also a pokemon type, built-in implies built into something, which doesn't sound quite right. Original implies that the rest are fake. I think standard is the best option.

**Comment by cclauss** [comment]
Nice work @samosaara ...
34 automated test cases are now running and only one is still failing...  https://travis-ci.org/cclauss/flake8_for_pt#L471  When doing `main.print_columns(Database().get_extras()` the names are in lowercase instead of title().
